x/leverage/keeper/reserves.go

```diff
@@ -28,31 +28,137 @@ func (k Keeper) GetReserveAmount(ctx sdk.Context, denom string) sdk.Int {
 	return amount
 }
 
-// IncreaseReserves adds an sdk.Coin (denom, amount) to the module's reserve
-// requirements.
-func (k Keeper) IncreaseReserves(ctx sdk.Context, coin sdk.Coin) error {
-	store := ctx.KVStore(k.storeKey)
-	if !k.IsAcceptedToken(ctx, coin.Denom) {
+// SetReserveAmount sets the amount reserved of a specified token.
+func (k Keeper) SetReserveAmount(ctx sdk.Context, coin sdk.Coin) error {
+	if !k.IsAcceptedToken(ctx, coin.Denom) || !coin.IsValid() {
 		return sdkerrors.Wrap(types.ErrInvalidAsset, coin.String())
 	}
 
-	// get current amount reserved for this asset type
+	store := ctx.KVStore(k.storeKey)
 	reserveKey := types.CreateReserveAmountKey(coin.Denom)
-	currentReserve := sdk.ZeroInt()
-
-	if bz := store.Get(reserveKey); bz != nil {
-		err := currentReserve.Unmarshal(bz)
-		if err != nil {
-			return err
-		}
-	}
 
-	// add the new reserve amount to the current one and save
-	bz, err := currentReserve.Add(coin.Amount).Marshal()
+	// save the new reserve amount
+	bz, err := coin.Amount.Marshal()
 	if err != nil {
 		return err
 	}
 
 	store.Set(reserveKey, bz)
 	return nil
 }
+
+// SweepBadDebts attempts to repay all bad debts in the system, efficiently
+// stopping for any denomination where reserves become exhausted. It should be
+// called every Interest Epoch.
+func (k Keeper) SweepBadDebts(ctx sdk.Context) error {
+	store := ctx.KVStore(k.storeKey)
+
+	// Outer iterator will iterate over all token denoms with bad debt
+	denomPrefix := types.CreateBadDebtDenomKeyNoDenom()
+	denomIter := sdk.KVStorePrefixIterator(store, denomPrefix)
+	defer denomIter.Close()
+
+	for ; denomIter.Valid(); denomIter.Next() {
+		// denomKey is prefix | denom
+		denomKey := denomIter.Key()
+		denom := string(denomKey[len(denomPrefix):]) // remove prefix
```

Comment:
With the 0x00 byte at the end : 
```suggestion
		denom := string(denomKey[len(denomPrefix):len(denomKey)-1]]) // remove prefix
```